CMakeLists.txt

```diff
@@ -150,22 +150,22 @@ endif()
 include(${HPCC_SOURCE_DIR}/cmake_modules/commonSetup.cmake)
 
 if ( PLUGIN )
-    HPCC_ADD_SUBDIRECTORY (common/deftype)
-    HPCC_ADD_SUBDIRECTORY (common/dllserver)
-    HPCC_ADD_SUBDIRECTORY (common/environment)
-    HPCC_ADD_SUBDIRECTORY (common/remote)
-    HPCC_ADD_SUBDIRECTORY (common/workunit)
-    HPCC_ADD_SUBDIRECTORY (ecl/hql)
-    HPCC_ADD_SUBDIRECTORY (fs/dafsclient)
+    HPCC_ADD_SUBDIRECTORY (common/deftype "NLP")
+    HPCC_ADD_SUBDIRECTORY (common/dllserver "NLP")
+    HPCC_ADD_SUBDIRECTORY (common/environment "NLP")
+    HPCC_ADD_SUBDIRECTORY (common/remote "NLP")
+    HPCC_ADD_SUBDIRECTORY (common/workunit "NLP")
+    HPCC_ADD_SUBDIRECTORY (ecl/hql "NLP")
+    HPCC_ADD_SUBDIRECTORY (fs/dafsclient "NLP")
     HPCC_ADD_SUBDIRECTORY (roxie/roxiemem)
     HPCC_ADD_SUBDIRECTORY (rtl/nbcd)
     HPCC_ADD_SUBDIRECTORY (rtl/eclrtl)
     HPCC_ADD_SUBDIRECTORY (system/jlib)
-    HPCC_ADD_SUBDIRECTORY (system/jhtree)
-    HPCC_ADD_SUBDIRECTORY (system/hrpc)
-    HPCC_ADD_SUBDIRECTORY (system/mp)
-    HPCC_ADD_SUBDIRECTORY (system/security/securesocket)
-    HPCC_ADD_SUBDIRECTORY (system/security/zcrypt)
+    HPCC_ADD_SUBDIRECTORY (system/jhtree "NLP")
+    HPCC_ADD_SUBDIRECTORY (system/hrpc "NLP")
+    HPCC_ADD_SUBDIRECTORY (system/mp "NLP")
+    HPCC_ADD_SUBDIRECTORY (system/security/securesocket "NLP")
+    HPCC_ADD_SUBDIRECTORY (system/security/zcrypt "NLP")
```

Comment:
These line changes were made to keep compile times down when we are building plugins. Those directories weren't needed to create the other plugins. If it's an NLP package, they'll get included in the build.  But I went through those directories and added guards on the install() functions so we don't add a bunch of libraries to NLP that are already in the base platform.